backbone.js

```diff
@@ -1487,11 +1486,12 @@
       this._wantsHashChange = this.options.hashChange !== false;
       this._hasHashChange   = 'onhashchange' in window;
       this._wantsPushState  = !!this.options.pushState;
-      this._hasPushState    = !!(this.options.pushState && this.history && this.history.pushState);
+      this._hasPushState    = !!(this.history && this.history.pushState);
+      this._usePushState    = this._wantsPushState && this._hasPushState;
       this.fragment         = this.getFragment();
 
-      // Normalize root to always include a leading and trailing slash.
-      this.root = ('/' + this.root + '/').replace(rootStripper, '/');
+      // Normalize root to always include a leading slash.
+      this.root = (this.root || '/').replace(/^[^\/]/, '/$&');
```

Comment:
I think this can be simpler. The default is already handled above. Also, we should remove `rootStripper` if it's not in use. I think we can avoid the regex entirely with the following.

```js
if (this.root.charAt(0) !== '/') this.root = '/' + this.root;
```